module/mempool/entity/executableblock.go

```diff
@@ -102,7 +102,7 @@ func (b *ExecutableBlock) HasAllTransactions() bool {
 // HasStartState returns whether the block has StartState, which
 // indicates whether its parent has been executed.
 func (b *ExecutableBlock) HasStartState() bool {
-	return len(b.StartState) > 0
+	return b.StartState != flow.EmptyStateCommitment
```

Comment:
⚠️ 
that is a tough assertion 😅 :
* Essentially it means, that we cannot have a genesis block with a completely empty state. 

Maybe it would be better to change 
https://github.com/onflow/flow-go/blob/00444a2404a18da1406bd989dbcf658bf152a4f9/module/mempool/entity/executableblock.go#L28
to a pointer type `*flow.StateCommitment` because it seems that it is an optional value. Then we would use `nil` as value not set (inset of associating an empty genesis state with "value not set").

Reply:
Good catch! This is an example of the errors I want us to avoid (it's happening in many places with identifiers for instance). 

I have implemented your suggestion in another branch https://github.com/onflow/flow-go/commit/f7f27c479fe3116c7f1d539cca1316aadc9e4fda. 

Reading the comments of `HasStartState()`, it says 
```// HasStartState returns whether the block has StartState, which indicates whether its parent has been executed.```
Using an array pointer, I understand we would embed two information in the `StartState` field:
 - whether the parent block has been executed (by comparing with `nil`)
 - the state value, in the case when the parent has been executed
 I think it is cleaner to separate these two information in two separate fields:
 ```
 ParentWasExecuted   bool
 StartState                   flow.StateCommitment
 ```
 This also avoid using the heap (pointers to arrays cause the data to move from the stack to the heap), and we maintain data locality.

Reply:
In the case `ParentWasExecuted` is `false`, `StartState` will get the dummy value. I don't think we should worry about "accidental" reads of that value or developers errors. That could happen anyway if we are using array pointers, when the state is referenced by a nil pointer (which panics btw).